python/lsst/daf/butler/datastores/file_datastore/get.py

```diff
@@ -225,7 +226,7 @@ def _read_artifact_into_memory(
 
     formatter = getInfo.formatter
     nbytes_max = 10_000_000  # Arbitrary number that we can tune
-    if resource_size <= nbytes_max and formatter.can_read_bytes():
+    if recorded_size >= 0 and recorded_size <= nbytes_max and formatter.can_read_bytes():
```

Comment:
Good catch. The size of -1 for "unknown" was added long after this check was written.